src/MetadataUtility/Audio/Wamd.cs

```diff
@@ -0,0 +1,203 @@
+            wamdOffset += 2;
+
+            uint length = BinaryPrimitives.ReadUInt32LittleEndian(wamdSpan[wamdOffset..]);
+            wamdOffset += 4;
+
+            if (subChunkId != 0 || length != 2)
+            {
+                return WamdVersionError;
+            }
+
+            ushort version = BinaryPrimitives.ReadUInt16LittleEndian(wamdSpan[wamdOffset..]);
+
+            return version;
+        }
+
+        /// <summary>
+        /// Parses a date into Nodatime's OffsetDateTime.
+        /// </summary>
+        /// <param name="value">The date to parse.</param>
+        /// <returns>The parsed date.</returns>
+        public static OffsetDateTime? DateParser(string value)
+        {
+            OffsetDateTime? date = null;
+
+            try
+            {
+                date = OffsetDateTimePattern.CreateWithInvariantCulture("yyyy'-'MM'-'dd' 'HH':'mm':'sso<m>").Parse(value).Value;
+            }
+            catch (UnparsableValueException)
+            {
+                return null;
+            }
+
+            return date;
+        }
+
+        /// <summary>
+        /// Parses a location's longitude and latitude.
+        /// Assigns the location data to a given wamd object.
+        /// </summary>
+        /// <param name="value">The location to parse.</param>
+        /// <param name="wamdData">The wamd object.</param>
+        public static void SetLocation(string value, Wamd wamdData)
+        {
+            string[] locationInfo = value.Split(",", StringSplitOptions.RemoveEmptyEntries);
+
+            double latitude = double.Parse(locationInfo[0]);
+            string latitudeDirection = locationInfo[1];
+
+            double longitude = double.Parse(locationInfo[2]);
+            string longitudeDirection = locationInfo[3];
+
+            wamdData.Latitude = latitudeDirection.Equals("N") ? latitude : latitude * -1;
+            wamdData.Longitude = latitudeDirection.Equals("E") ? longitude : longitude * -1;
+        }
+
+        /// <summary>
+        /// Extracts metadata from a wamd chunk.
+        /// </summary>
+        /// <param name="wamdSpan">The wamd chunk.</param>
+        public static Wamd ExtractMetadata(ReadOnlySpan<byte> wamdSpan)
+        {
+            Wamd wamdData = new Wamd();
+
+            // Links metadata set functions to their corresponding subchunk ID
+            Dictionary<int, Action<string>> setters = new Dictionary<int, Action<string>>
+            {
+                { 1, value => wamdData.Name = value },
+                { 2, value => wamdData.SerialNumber = value },
+                { 3, value => wamdData.Firmware = value },
+                { 5, value => wamdData.StartDate = DateParser(value) },
+                { 18, value => wamdData.MicrophoneType = value.Split(",") },
+                { 19, value => wamdData.MicrophoneSensitivity = value.Split(",") },
+                { 20, value => SetLocation(value, wamdData) },
+                { 21, value => wamdData.Temperature = value },
```

Comment:
Some points:  

- make an explicit effort to name those magic numbers (and also format them as hex over decimal) as they are in the wamd header file - makes it much easier to cross reference them
- You've got a mix between mutative and pure functions doing the parsing? Why? I'd probably lean always towards Pure to isolate effects
- This dictionary can be statically defined once at the class level. Defining it in this method means it is always defined on every invocation of the method.
- And why the partial implementation?

Reply:
By partial implementation did you just mean I'm not accounting for every value in their metadata doc? Pulled what I thought were all the important metadata that I saw in the files I have. Didn't think anything else in there correlated to metadata fields that we have. The only ones that could be interesting but weren't in any of the files I've seen are humidity or light? Could add metadata fields for those, or implement others if there are any you think are of value?

Reply:
You took a practical approach. I prefer myself to create fully functional primitives.

One eventual goal of EMU is to be able to strip any metadata out of any file and then allow said files to be converted without loss of information. That might be a good reason to fully deserialize the chunk.

But if that doesn't align with current goals I'm happy if you leave it until later.